src/ansys/dpf/core/animator.py

```diff
@@ -152,6 +152,130 @@ def animation():
         self._plotter.close()
         return result
 
+    def animate_workflow_modal(
```

Comment:
I think the a better option would be to reuse animate_workflow above, modifying it if required.
I think if the scaling and the "indices" are given as "loop_over" and "scale_factor" to animate_workflow, it will already be near what is intended.